feat: add kwargs to paginator send & reply methods

SamNuttall opened the pull request:
## Pull Request Type
<!-- Check the appropriate option -->

- [x] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description
Adds keyword arguments to paginator send and reply methods. 
These are then passed to the respective methods to pass-through their functionality

Example of usage - sending an ephemeral paginator: 
```
paginator = Paginator.create_from_string(bot, your_content, page_size=1000)
await paginator.send(ctx, ephemeral=True)  # <-- Added ephermal=True (other code taken from docs)
```

## Changes

- Added kwargs to paginator send & reply methods
- Updated BaseContext to InteractionContext as BaseContext does not include a send method


## Related Issues
<!-- If this PR is related to any open issues, please mention them using "#ISSUENUMBER" -->


## Test Scenarios
<!-- Provide clear instructions on how to test your changes, where applicable - if no tests are required, explain why -->


## Python Compatibility
<!-- Testing 3.11 is not strictly required, but it would be nice if you could confirm that it works. -->
- [ ] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`


## Checklist
<!-- If you have not completed all of the following, there is a good chance your PR will not be merged. -->
- [X] I've run the `pre-commit` code linter over all edited files
- [ ] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [X] I've updated / added  documentation, where applicable

SamNuttall committed:
feat: add kwargs to send and reply in paginator

This allows paginators to be sent ephemerally by passing kwargs to paginator methods

SamNuttall committed:
style: add whitespace below args (autofix by pre-commit)

LordOfPolls commented on interactions/ext/paginators.py:
```diff
@@ -352,18 +352,19 @@ def to_dict(self) -> dict:
         }
 
-    async def send(self, ctx: BaseContext) -> Message:
+    async def send(self, ctx: InteractionContext, **kwargs) -> Message:
```
```suggestion
    async def send(self, ctx: BaseContext, **kwargs) -> Message:
```

LordOfPolls commented on interactions/ext/paginators.py:
```diff
@@ -13,7 +13,7 @@
     spread_to_rows,
     ComponentCommand,
-    BaseContext,
+    InteractionContext,
```
```suggestion
    BaseContext,
```

LordOfPolls reviewed (changes requested):
Reverted your typehint change. While yes, BaseContext is missing the send method, this is an issue with that object and should not be changed in this PR

Once this is changed, im happy for this to be merged

SamNuttall committed:
refactor: revert BaseContext typehint change

Co-authored-by: LordOfPolls <dev@lordofpolls.com>
Signed-off-by: Sam Nuttall <dev@samln.dev>

pre-commit-ci[bot] committed:
ci: correct from checks.

SamNuttall committed:
fix: re-add BaseContext dependancy

InteractionContext auto removed by pre-commit

SamNuttall commented:
Reverted the typehint change as requested - I'll take a further look into the context types so I know in future :)

Do note that your second suggestion is now outdated as pre-commit removed the InteractionContext line, though I have fixed this via a manual commit.
